app/serializers/base_bookmarkable.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+##
+# Anything that we want to be able to bookmark must be registered as a
+# bookmarkable type using Bookmark.register_bookmarkable(bookmarkable_klass),
+# where the bookmarkable_klass is a class that implements this BaseBookmarkable
+# interface. Some examples are TopicBookmarkable and PostBookmarkable.
+#
+# These methods are then called by the Bookmarkable class through a public
+# interface, used in places where we need to list, send reminders for,
+# or otherwise interact with bookmarks in a way that is unique to the
+# bookmarkable type.
+class BaseBookmarkable
+  attr_reader :model, :serializer, :preload_associations
+
+  def initialize(model, serializer)
+    @model = model
+    @serializer = serializer
+  end
+
+  def list_query(user, guardian)
```

Comment:
Since we're designing an interface here, I think we need to include documentation on what types the params are and what return value is expected.

Reply:
It's weird because it's never called directly -- the `Bookmarkable` class calls these, and it already has documentation for all the parameters, so I feel like it would be doubling up. I can add return value documentation there -- I already have a note for `BaseBookmarkable` to see `Bookmarkable` for this kind of thing. I want everything going through `Bookmarkable` in case we need to add additional code which is _not_ defined by the bookmarkable type class, e.g. like this:

https://github.com/discourse/discourse/blob/fdb440f08b0a19d2ad4a15ed78650281b75f9d90/app/services/bookmarkable.rb#L55-L59

Reply:
I suppose...maybe I will just move the docs to `BaseBookmarkable` and leave `Bookmarkable` with minimal docs and point to `BaseBookmarkable` for more thorough details then :+1:

Reply:
Done